tests/meta/clean/conftest.py

```diff
@@ -1,59 +1,207 @@
-    return correct_flow_cell_name + "_r"
+def flow_cell_clean_api_can_not_be_removed(
+    tmp_flow_cell_not_to_clean_path: Path,
+    store_with_flow_cell_not_to_clean: Store,
+    real_housekeeper_api: HousekeeperAPI,
+) -> CleanFlowCellAPI:
+    return CleanFlowCellAPI(
+        flow_cell_path=tmp_flow_cell_not_to_clean_path,
+        status_db=store_with_flow_cell_not_to_clean,
+        housekeeper_api=real_housekeeper_api,
+        dry_run=False,
+    )
 
 
 @pytest.fixture(scope="function")
-def incorrect_flow_cell_extension(correct_flow_cell_name: str) -> str:
-    """Incorrect flow cell name."""
-    return correct_flow_cell_name + ".someextension"
+def tmp_flow_cell_to_clean_path(tmp_flow_cell_directory_bclconvert: Path):
+    """Returns the path to a flow cell fulfilling all cleaning criteria."""
+    return tmp_flow_cell_directory_bclconvert
 
 
 @pytest.fixture(scope="function")
-def incorrect_flow_cell_name() -> str:
-    """Incorrect flow cell name."""
-    return "201203_A00689_0200_AZZZZZZZZZ"
+def tmp_flow_cell_to_clean(tmp_flow_cell_to_clean_path: Path) -> FlowCellDirectoryData:
+    """Returns a flow cell directory object for a flow cell that fulfills all cleaning criteria."""
+    return FlowCellDirectoryData(tmp_flow_cell_to_clean_path)
 
 
 @pytest.fixture(scope="function")
-def correct_flow_cell_path(
-    tmp_demultiplexed_runs_directory: Path, correct_flow_cell_name: str
-) -> Path:
-    """Full path to a correctly named flow cell directory in demultiplexed-runs."""
-    return Path(tmp_demultiplexed_runs_directory, correct_flow_cell_name)
+def tmp_flow_cell_not_to_clean_path(tmp_flow_cell_directory_bcl2fastq: Path):
+    """Return the path to a flow cell not fulfilling all cleaning criteria."""
+    return tmp_flow_cell_directory_bcl2fastq
 
 
 @pytest.fixture(scope="function")
-def incorrect_flow_cell_path_too_long(
-    tmp_demultiplexed_runs_directory: Path, incorrect_flow_cell_too_long: str
-) -> Path:
-    """Full path to an incorrectly named flow cell directory in demultiplexed-runs."""
-    return Path(tmp_demultiplexed_runs_directory, incorrect_flow_cell_too_long)
+def tmp_flow_cell_not_to_clean(tmp_flow_cell_not_to_clean_path: Path) -> FlowCellDirectoryData:
+    """Returns a flow cell directory object for a flow cell that does not fulfill all cleaning criteria."""
+    return FlowCellDirectoryData(tmp_flow_cell_not_to_clean_path)
+
+
+@pytest.fixture(scope="session")
+def tmp_sample_sheet_clean_flow_cell_path(tmp_path_factory) -> Path:
+    sample_sheet_path = tmp_path_factory.mktemp("SampleSheet.csv")
+    return sample_sheet_path
+
+
+@pytest.fixture
+def store_with_flow_cell_to_clean(
+    store: Store,
+    sample_id: str,
+    tmp_flow_cell_to_clean: FlowCellDirectoryData,
+    helpers: StoreHelpers,
+) -> Store:
+    """Return a store with multiple samples with sample lane sequencing metrics."""
+    sample_sequencing_metrics_details: List[Union[str, str, int, int, float, int]] = [
```

Comment:
Would it be possible to make this into a data class or a named tuple?

Reply:
I think that would be possible. There is another fixture using a similar approach. Maybe we can fix both of them at the same time in another PR.

Reply:
Is this how `Union` works? I think you just add the possible types the entries, not the type of each entry. So it would be `Union[str, int, float]`

Reply:
But I agree, a class would be great if is not too much work